src/js/player.js

```diff
@@ -823,6 +825,179 @@ class Player extends Component {
   }
 
   /**
+   * Modify the video/audio element so that we can detect when
+   * the source is changed. Fires `sourceset` just after the source has changed
+   *
+   * @fires Player#sourceset
+   */
+  watchForSourceSet_() {
+    // if we cannot overwrite the src property, there is no support
+    // iOS 7 safari for instance cannot do this.
+    try {
+      const el = document.createElement('video');
+
+      Object.defineProperty(el, 'src', {
+        get() {},
+        set() {}
+      });
+    } catch (e) {
+      return;
+    }
+
+    const el = this.el().getElementsByTagName('video')[0] || this.el().getElementsByTagName('audio')[0];
+
+    if (!el) {
+      if (!this.isReady_) {
+        this.ready(() => this.watchForSourceSet_());
+      }
+      return;
+    }
+
+    // we need to fire sourceset when the player is ready
+    // if we find that the media element had a src when it was
+    // given to us
+    if (this.tagAttributes && this.tagAttributes.src) {
+      this.ready(() => {
+        this.trigger('sourceset');
+      });
+    }
+
+    const proto = window.HTMLMediaElement.prototype;
+    const srcDescriptor = {
+      get() {
+        return proto.getAttribute.call(this, 'src');
+      },
+      set(v) {
+        return proto.setAttribute.call(this, 'src', v);
+      },
+      enumerable: true
+    };
+
+    // preserve getters/setters already on `el.src` if they exist
+    if (Object.getOwnPropertyDescriptor(el, 'src')) {
+      Object.assign(srcDescriptor, Object.getOwnPropertyDescriptor(el, 'src'));
+    } else if (Object.getOwnPropertyDescriptor(proto, 'src')) {
+      Object.assign(srcDescriptor, Object.getOwnPropertyDescriptor(proto, 'src'));
+    }
+
+    Object.defineProperty(el, 'src', {
+      get: srcDescriptor.get.bind(el),
+      set: (v) => {
+        const retval = srcDescriptor.set.call(el, v);
+
+        this.ready(() => this.trigger('sourceset'), true);
+
+        return retval;
+      },
+      configurable: true,
+      enumerable: srcDescriptor.enumerable
+    });
+
+    const oldSetAttribute = el.setAttribute;
+
+    el.setAttribute = (n, v) => {
+      const retval = oldSetAttribute.call(el, n, v);
+
+      if (n === 'src') {
+        this.ready(() => this.trigger('sourceset'), true);
+      }
+
+      return retval;
+    };
+
+    const oldLoad = el.load;
+
+    el.load = () => {
+      const retval = oldLoad.call(el);
+
+      this.ready(() => this.trigger('sourceset'), true);
+
+      return retval;
+    };
+
+    this.on('dispose', () => {
+      if (!el) {
+        return;
+      }
```

Comment:
When would this condition be `true`?

Reply:
I don't think it ever will be, but I just assume anything is possible these days haha. I don't feel too strongly about removing it, but I would prefer to just comment that we don't think it will run.